doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -766,58 +686,161 @@ the conversion in hypotheses :n:`{+ @ident}`.
+apply unfolding strategies such as from :cmd:`Strategy`.
 
-.. tacv:: pattern {+, @term {? at {? -} {+, @natural}}}
+:tacn:`native_compute` is typically two to five
+times faster than :tacn:`vm_compute` at applying conversion rules
+when Coq is running native code, but :tacn:`native_compute` required
+considerably more overhead.  We recommend using :tacn:`native_compute`
+when all of the following are true (otherwise use :tacn:`vm_compute`):
 
-   This is the most general syntax that combines the different variants.
+- the running time in :tacn:`vm_compute` at least 5-10 seconds
+- the size of the input term is small (e.g. hand-generated code rather than
+  automatically-generated code that may have nested destructs on inductives
+  with dozens or hundreds of constructors)
+- the output is small (e.g. you're returning a boolean, a natural number or
+  an integer rather than a large abstract syntax tree)
+
+
+.. tacn:: vm_compute {? {| @reference_occs | @pattern_occs } } {? @occurrences }
+
+   Evaluates the goal using the optimized call-by-value evaluation
+   bytecode-based virtual machine described in :cite:`CompiledStrongReduction`.
+   This algorithm is dramatically more efficient than the algorithm used for the
+   :tacn:`cbv` tactic, but it cannot be fine-tuned. It is especially useful for
+   full evaluation of algebraic objects. This includes the case of
+   reflection-based tactics.
+
+.. tacn:: native_compute {? {| @reference_occs | @pattern_occs } } {? @occurrences }
+
+   Evaluates the goal by compilation to OCaml as described
+   in :cite:`FullReduction`. Depending on the configuration, this tactic can either default to
+   :tacn:`vm_compute`, recompile dependencies or fail due to some missing
+   precompiled dependencies,
+   see :ref:`the native-compiler option <native-compiler-options>` for details.
+
+   .. flag:: NativeCompute Timing
+
+      This flag causes all calls to the native compiler to print
+      timing information for the conversion to native code,
+      compilation, execution, and reification phases of native
+      compilation.  Timing is printed in units of seconds of
+      wall-clock time.
+
+   .. flag:: NativeCompute Profiling
+
+      On Linux, if you have the ``perf`` profiler installed, this flag makes
+      it possible to profile :tacn:`native_compute` evaluations.
+
+   .. opt:: NativeCompute Profile Filename @string
+
+      This option specifies the profile output; the default is
+      ``native_compute_profile.data``. The actual filename used
+      will contain extra characters to avoid overwriting an existing file; that
+      filename is reported to the user.
+      That means you can individually profile multiple uses of
+      :tacn:`native_compute` in a script. From the Linux command line, run ``perf report``
+      on the profile file to see the results. Consult the ``perf`` documentation
+      for more details.
+
+Computing in a term: eval
```

Comment:
I realize that this is the title of the subsection that you moved from the Tactics chapter, but I would modify it slightly to also include `Eval`.

Reply:
Computing in a term: eval and Eval